ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/BlockchainQueries.java

```diff
@@ -176,6 +206,20 @@ public long headBlockNumber() {
     return fromAccount(address, blockNumber, Account::getCode, Bytes.EMPTY);
   }
 
+  /**
+   * Retrieves the code associated with the given account at a particular block hash.
+   *
+   * @param address The account address being queried.
+   * @param blockHash The hash of the block to be checked.
+   * @return The code associated with this address.
+   */
+  public Optional<Bytes> getCode(final Address address, final Hash blockHash) {
+    final long blockNumber =
```

Comment:
Again, we lose orphan hash data going to a number and using that.  The most likely scenario for failure is that a contract was deployed on an orphan chain but not the main chain, or vice versa, because the TXes committed were different.

We can't simply do a hash->BlockNumber query here.  What if they are querying from an orphaned block hash?  The numbers will be wrong.

`getCode` and the whole call series needs to be primarily hash based, where the calls to get it by block number quickly go to the hash based call chain.